### PR TITLE
fix: SDK Defaults Config - Handle config injection for None Sessions

### PR DESCRIPTION
**Pull request by rubanh**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

TBA: unit tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/utils.py [hunk 1]
```diff
@@ -1084,23 +1096,34 @@ def resolve_value_from_config(
...
 
     Args:
         key: Key Path of the config file entry.
         sagemaker_session (sagemaker.session.Session): A SageMaker Session object, used for
-        SageMaker interactions.
+            SageMaker interactions.
+        sagemaker_config (dict): The sdk defaults config that is normally accessed through a
+            Session object by doing `session.sagemaker_config`. (default: None) This parameter will
+            be checked for the config value if (and only if) sagemaker_session is None. This
+            parameter exists for the rare cases where no Session provided no Session but a default
```
Comment (rubanh) [inline src/sagemaker/utils.py:1109]: small typo

File: tests/unit/test_utils.py [hunk 2]
```diff
@@ -1239,6 +1251,38 @@ def test_resolve_value_from_config():
...
+    sagemaker_session.sagemaker_config.update(
+        {"SageMaker": {"EndpointConfig": {"KmsKeyId": "CONFIG_VALUE"}}}
+    )
+    sagemaker_config = {
+        "SchemaVersion": "1.0",
+        "SageMaker": {"EndpointConfig": {"KmsKeyId": "CONFIG_VALUE"}},
+    }
+
+    # Tests that the function returns the correct value when the key exists in the sagemaker_session configuration.
+    assert get_sagemaker_config_value(sagemaker_session, config_key_path) == "CONFIG_VALUE"
```
Comment (rubanh) [inline tests/unit/test_utils.py:1272]: nit: should we use named args for clarity?

File: tests/unit/test_utils.py [hunk 3]
```diff
@@ -1239,6 +1251,38 @@ def test_resolve_value_from_config():
     mock_info_logger.reset_mock()
 
 
+def test_get_sagemaker_config_value():
+    mock_config_logger = Mock()
+
+    mock_info_logger = Mock()
+    mock_config_logger.info = mock_info_logger
```
[the remark this answers is not in the document]
Comment (knikure) [reply tests/unit/test_utils.py:1258]: I saw that no unit test was present to test get_sagemaker_config_value() function so I added this. It is better to have all functions covered through unit-tests

File: src/sagemaker/utils.py (hunk 1 above)
Comment (knikure) [reply src/sagemaker/utils.py:1109]: done

File: src/sagemaker/jumpstart/utils.py [hunk 4]
```diff
@@ -455,6 +456,10 @@ def resolve_model_intelligent_default_field(
     over intelligent defaults. For all other fields, intelligent defaults takes precedence
     over the JumpStart default fields.
     """
+    # Workaround for config injection if sagemaker_session is None, since in
+    # that case sagemaker_session will not be initialized until
+    # `_init_sagemaker_session_if_does_not_exist` is called later
```
[the remark this answers is not in the document]
Comment (knikure) [reply src/sagemaker/jumpstart/utils.py:461]: updated comment

**Review by akrishna1995 — APPROVED** [review]

**Pull request merged**